### ISSUE TITLE
Support Angular 2.0.0-rc.3

### ISSUE THREAD

**Issue by fuitattila** [dougludlow/ng2-bs3-modal#68]
Dear Doug!

Can you please upgrade your module to support Angular 2.0.0-rc.3?

https://github.com/angular/angular/blob/master/CHANGELOG.md#200-rc3-2016-06-21

Or your package works properly with rc,3? What do you think?

Thank you!

**Comment by dougludlow** [comment]
Hey @fuitattila. I'm not sure if it works with rc.3 or not. I currently don't have projects that are actively using it. I'm a little tied up at the moment, but hopefully I can test it out within the next week or so.

**Comment by fuitattila** [comment]
Anyway I will try it soon - and I send for you my expereience here. Thank you Doug!

**Comment by kenkopelson** [comment]
Angular 2 is at rc4, and rc5 is quickly coming.  Please update to the newest version of Angular.

**Comment by dougludlow** [comment]
Yeah... sorry guys, I've been in the middle of moving... anyone can feel free to put in a PR for it.

**Comment by samvloeberghs** [comment]
on it, will do upgrade for rc4

**Comment by samvloeberghs** [comment]
I managed to update to RC4 and the new router:
https://github.com/samvloeberghs/ng2-bs3-modal

But I need some more time to get the tests working

**Comment by samvloeberghs** [comment]
@dougludlow 
I've got 5 tests running but having troubles with the 3 (xit).
Perhaps you could take a look at this:

https://github.com/samvloeberghs/ng2-bs3-modal

**Comment by dougludlow** [comment]
@samvloeberghs I'm a bit busy atm, but i'll take a look when I get the chance. As new releases come out, I've typically been looking at the angular2 source tests and modelling mine after theirs. For example: https://github.com/angular/angular/blob/master/modules/%40angular/router/test/router.spec.ts

**Comment by koleary94** [comment]
hey @dougludlow and @samvloeberghs. I just opened up a pr that should complete @samvloeberghs changes to the angular rc4 upgrade.

Not sure how you guys wanted to handle merging the changes, but I figured forking off of @samvloeberghs and submitting a pr directly to the main master worked best.

@samvloeberghs just in case you're curious why the other tests weren't working, not using the `.then()` promise syntax was causing race conditions with opening and closing the modal, which is why there were timeouts. Also, the routing test needed access to the app component, so the url routing could actually occur. (There could have been an alternative solution that didn't involve instantiating it, but that's what I found worked best). 

Also, @samvloeberghs here's the [diff](https://github.com/dougludlow/ng2-bs3-modal/pull/82/files?diff=unified) between our code
